### PR TITLE
doc: release: NXP contributions

### PR DESCRIPTION
**Pull request by yvanderv**
Adding NXP's contributions to v4.0

### CODE REVIEW

**Review by JarmouniA — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:607]: remove this (see #81242)

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:511]: remove these ethernet notes see #81242

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:687]: missing spi changes

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:554]: not sure if this needs to go in the release notes as it isn't exposed to users, it's basically just a refactor of where some code was in the tree

**Review by decsny — DISMISSED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (yvanderv) [reply doc/releases/release-notes-4.0.rst:68]: Needed clarification (arm_mpu_v8)

**Comment by yvanderv** [comment]
@jarmounia, your change suggestions have been integrated. Thanks.

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:687]: What are the spi changes that are missing?

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:416]: definitely not something worth mentioning in release notes :) drop the entire change since you're also introducing a duplicated bullet item with "Added two new build ..."

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:451]: introduction of SCMI in general probably deserves its own entry (under "Architectures")? @LaurentiuM1234

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:427]:
```suggestion
  * Added support for NXP S32 SAR_ADC (:dtcompatible:`nxp,s32-adc-sar`).
```

in general, it would be nice to add references to the compatible whenever mentioning additions/changes to drivers in your PR

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:748]: you should really focus on changes that are related to NXP Platforms, not changes contributed by NXP employees, otherwise you're basically doing the work of other areas' maintainers and, more importantly, going to get yourself into lots of merge conflicts if other PRs (see e.g. https://github.com/zephyrproject-rtos/zephyr/pull/81216) get merged before yours..

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1009]: this is already covered in the Video section..

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1310]: again, likely already covered in https://github.com/zephyrproject-rtos/zephyr/pull/81216

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1349]: as before, don't go to the level of details where you're describing changes or tweaks being made to tests (to put it differently: if the release notes are effectively going to the level of detail were it's one bullet point per PR or commit, then we'd rather point users directly to the GitHub interface to see what's new by themselves :) )

**Review by kartben — DISMISSED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:687]: unless I'm missing something, I don't see any notes on spi changes in this PR

File: doc/releases/release-notes-4.0.rst
Comment (LaurentiuM1234) [reply doc/releases/release-notes-4.0.rst:451]: hmm, or maybe have it under "Libraries/Subsystems" section as its own subsection or add a "Firmware" section under "Libraries/Subsystems" and have it there? but then again the main idea is to inform people that Zephyr has SCMI support so I'm fine with having it anywhere as long as it makes sense.

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:748]: yeah, this point was brought up many times internally, I don't know why we are still doing this

**Review by JarmouniA — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [reply doc/releases/release-notes-4.0.rst:214]: Ya, @yvanderv to be clear and to echo this comment and mine from earlier, I didn't really bother commenting on all the changes that are not release note worthy and I hope you will be making a global cleanup that goes beyond the couple specific comments that were left by @jarmouniA and I

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:172]: Agreed

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:312]: As already mentioned, this needs to be removed.
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-4.0.rst?plain=1#L272

Please try to re-focus your PR on changes to NXP platforms, not just any commit from NXP as this unmanageable otherwise

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:214]: There was a time where we would run a script to add to the release notes all bugs that were fixed. A request to clean up items that are "not release note worthy" is a bit subjective. @kartben, do we have any guidance on what is considered "release note worthy"?

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:284]: Agreed

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:501]: was that in this PR or a separate one?

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:715]: @yvanderv, I did a quick review of #81216 and it appears Jukkar generally captured the jist of these. We can remove them here.

File: doc/releases/release-notes-4.0.rst
Comment (dleach02) [reply doc/releases/release-notes-4.0.rst:1276]: @jukkar Your PR didn't add a HOSTAP section but instead captured some (or most of this) under WiFi. Is this what you want to do going forward?

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [reply doc/releases/release-notes-4.0.rst:214]: Right but that script's output was an *addition* to the already manually curated release notes, and it was abandoned as it was adding no value compared to what one would get by querying GitHub or grepping the commits. I believe the intent of the release notes has always been to be a *summary* of the "new and noteworthy" for a given release, so we have to draw a line somewhere to not end up with thousands of bullet points that are impossible to read through.

I agree it's somewhat subjective but looking at the level of details from other contributions made to the release notes so far gives a good sense of the level of detail typically targeted. Most, if not all, vendors made contributions that fit in a dozen bullet points (sometimes a couple dozen)

As an example, detailing changes being made to tests such as "Tailored the SPI_LOOPBACK test to the limitations of NXP Kinetis MCU's" is likely "too much"

Comment (decsny) [reply doc/releases/release-notes-4.0.rst:214]: This one is a bit arguable, because I guess maybe if someone was using DT_PATH macro or something to get one of these nodes from DT, it would have to change, but this bullet doesn't explain really that , but yeah maybe it's pretty minor

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:189]: I would capitalize Devicetree

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:654]: there was at least the change that fixed the lpspi native (not gpio) chip selelct when using syncrhonous DMA api on MCXN, that might be worth mentioning

File: doc/releases/release-notes-4.0.rst
Comment (jukkar) [reply doc/releases/release-notes-4.0.rst:1276]: @dleach02 I do not see much benefit to split the release note even more to have `Hostap` topic as it is part of Wi-Fi anyway.

**Review by JarmouniA — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (LaurentiuM1234) [reply doc/releases/release-notes-4.0.rst:423]: can drop this. Submitted https://github.com/zephyrproject-rtos/zephyr/pull/81330 for the SCMI notes.

File: doc/releases/release-notes-4.0.rst
Comment (yvanderv) [reply doc/releases/release-notes-4.0.rst:697]: well, why punctuation specifically here where it's absent all over the place?
Proposal: we make it an explicit rule and a linter is added.

**Review by JarmouniA — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst
Comment (yvanderv) [reply doc/releases/release-notes-4.0.rst:149]: A mystery to me too.

**Review by JarmouniA — APPROVED** [review]
@yvanderv Looking good, thanks!

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:224]: another thing that appeared after 4.0 dev cycle started, no need to report

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:230]: wasn't this done for every platform in the tree?

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:325]: double space in this line

Comment (decsny) [reply doc/releases/release-notes-4.0.rst:325]: also which SOC are these

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:328]: wasn't this also another effort done for all boards in tree, not just nxp

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:326]: the board itself was also new

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:233]: the whole SOC was added not just lpspi ?

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:815]: not fixed yet

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:1489]: why is this section being added

File: doc/releases/release-notes-4.0.rst
Comment (yvanderv) [reply doc/releases/release-notes-4.0.rst:211]: OK

Comment (decsny) [reply doc/releases/release-notes-4.0.rst]: this looks to still be in the PR

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:222]: not changed

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:228]: im pretty sure that not only was this done for every platform in tree but also the entire  MCX C was added during this development cycle 

#78235
#76865

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:327]: it looks like in the pinctrl driver section of these release notes, its mentioned but with only STM32: in front of the bullet, maybe just delete that STM32: before the line instead of making this new bullet under nxp platform

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:325]: do we need to document that we added documentation

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst:326]: can we clarify somewhere that this board was added new or is that already somewhere else

Comment (decsny) [reply doc/releases/release-notes-4.0.rst:326]: nevermind, looks like it's somewhere else

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [inline doc/releases/release-notes-4.0.rst:192]: where is MCXW71 SOC enablement mentioned

File: doc/releases/release-notes-4.0.rst
Comment (yvanderv) [reply doc/releases/release-notes-4.0.rst:325]: It is one addition that wasn't there before, it contains useful information

File: doc/releases/release-notes-4.0.rst
Comment (decsny) [reply doc/releases/release-notes-4.0.rst]: is this for MCXW and MCXN ?

**Comment by mmahadevan108** [comment]
@decsny @kartben , kindly re-visit this PR.

**Comment by decsny** [comment]
> @decsny @kartben , kindly re-visit this PR.

i reviewed it 10 minutes before this post and my comments still arent addressed, the whole mention of MCXC enablement is missing still for example, and multiple other things still need fixed

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [reply doc/releases/release-notes-4.0.rst:326]: I agree, please remove it. It doesn't make sense to get to this level of detail. Again, please take inspiration from other contributions already made to the release notes or from previous contributions such as https://github.com/zephyrproject-rtos/zephyr/pull/69100/files.
I by no means want to minimize NXP's contributions to the release but the level of detail in this PR is really not helping keep the release notes easy to read, and I'm afraid it's a bit counterproductive that for wanting to highlight too much you will in fact highlight nothing for the reader will just be deterred.

**Comment by yvanderv** [comment]
> > @decsny @kartben , kindly re-visit this PR.
> 
> i reviewed it 10 minutes before this post and my comments still arent addressed, the whole mention of MCXC enablement is missing still for example, and multiple other things still need fixed

This has been addressed

**Comment by yvanderv** [comment]
@mmahadevan108 @danieldegrasse @kartben , please review.

File: doc/releases/release-notes-4.0.rst
Comment (danieldegrasse) [inline doc/releases/release-notes-4.0.rst:441]: Sorry I missed this- these look like they should be in the "Counter" section, not DAC

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:798]: this is already mentioned in the nxp section

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1501]: this is really not that important

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1504]: not something for release notes

File: doc/releases/release-notes-4.0.rst
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:1508]: sounds like a really minor change , not sure why this is mentioned?

**Comment by kartben** [comment]
I will be dismissing my review so that this can be approved by others if they feel this is good enough.

**Review by mmahadevan108 — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by dleach02** [comment]
Mahesh is going to follow up with a PR to address the last comments

**Pull request merged**
